stl/inc/thread

```diff
@@ -240,6 +254,93 @@ struct hash<thread::id> {
         return _Hash_representation(_Keyval._Id);
     }
 };
+
+#if _HAS_CXX20
+class jthread {
+public:
+    using id                 = thread::id;
+    using native_handle_type = thread::native_handle_type;
+
+    jthread() noexcept : _Impl{}, _Ssource{nostopstate} {}
+
+    template <class _Fn, class... _Args, enable_if_t<!is_same_v<_Remove_cvref_t<_Fn>, jthread>, int> = 0>
+    explicit jthread(_Fn&& _Fx, _Args&&... _Ax) {

```

Comment:
I believe we already issue that diagnostic when `_Start` constructs `_Decay_copied`?

Reply:
I think those requirements should be the same as the `is_constructible<decay_t` parts and this was damaged as part of the 'mandating the standard library' papers. I'll file an issue rather than changing the product here.

Reply:
In theory, we could be passing an lvalue of a copyable-but-not-movable abomination type. Our `is_constructible_v<T, const T&>` requirement is satisfied, but `is_move_constructible_v<T>` is `false` and the Standard wants us to diagnose the program as ill-formed.

FWIW: I vaguely recall trying and failing to convince LWG that the move requirements are extraneous in a review of the Mandates paper. IIRC libstdc++ does perform moves in addition to the decay-copy?

Reply:
I filed an issue saying that this was in error.

Reply:
https://cplusplus.github.io/LWG/issue3476

JW already commented that his implementation has no objections. Now only libc++ has the issue and I note that despite the move constructible requirements, nothing in here allows an implementation to actually perform that extra move (it says the _DECAY_COPY happens in the launching thread and the invoke happens in the launched thread, no mention of a move in the middle).